app/src/main/java/com/a0xffffffff/dinesum/MainActivity.java

```diff
@@ -103,25 +112,39 @@ private void initView() {
     }
 
     private void initData() {
-        fragments = new ArrayList<>(3);
-        items = new SparseIntArray(3);
+        fragments = new ArrayList<>(4);
+        items = new SparseIntArray(4);
 
-        Fragment userFragment = MainFragment.newInstance("User");
-        Fragment homeFragment = MainFragment.newInstance("Home");
-//        Fragment addFragment = MainFragment.newInstance("Add");
-        Fragment addFragment = NewRequestFragment.newInstance();
+        mRequesterFragment = MainFragment.newInstance("Requester");
+        mAcceptorFragment = MainFragment.newInstance("Acceptor");
+        mHomeFragment = RequestFeedFragment.newInstance();
+        mAddFragment = NewRequestFragment.newInstance();
 
-        fragments.add(userFragment);
-        fragments.add(homeFragment);
-        fragments.add(addFragment);
+        fragments.add(mRequesterFragment);
+        fragments.add(mAcceptorFragment);
+        fragments.add(mHomeFragment);
+        fragments.add(mAddFragment);
 
-        items.put(R.id.menu_user, 0);
-        items.put(R.id.menu_home, 1);
-        items.put(R.id.menu_add, 2);
+        items.put(R.id.menu_requester, 0);
+        items.put(R.id.menu_acceptor, 1);
+        items.put(R.id.menu_home, 2);
+        items.put(R.id.menu_add, 3);
 
         mViewPager.setAdapter(new VpAdapter(getFragmentManager(), fragments));
     }
 
+    public void onNearbyRequestsReady() {
+        mHomeFragment.initListView();
+    }
+
+    public void onRequesterRequestsReady() {
+//        mRequesterFragment.initListView();

```

Comment:
Why is the definition empty?

Reply:
Haven't done other fragments yet, other than RequestFeedFragment. This will be uncommented later when those are finished.